### PR TITLE
feat: expose dns resources and enable own rootHostedZone usage

### PR DESCRIPTION
**Pull request by flochaz**
* Expose `rootHostedZone` and `stagesHostedZones`.
* Enable to pass its own rootHostedZoneId

#89 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts [hunk 1]
```diff
@@ -250,6 +253,20 @@ test("should not create any zone if no domain is provided", () => {
...
+    new Stack(),
+    "AWSOrganizationsStack",
+    {
+      ...awsOrganizationsStackProps,
+      rootHostedZoneDNSName: "yourdomain.com",
+      existingRootHostedZoneId: "existing-root-zone-id"
+    }
+  );
+
+  expect(awsOrganizationsStack).toCountResources("AWS::Route53::HostedZone", 3);
```
Comment (jeromevdl) [inline source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts:267]: can we explicitly check there's 0 root hosted zone ?

File: source/aws-bootstrap-kit/lib/dns.ts [hunk 2]
```diff
@@ -42,6 +48,7 @@ export class RootDns extends Construct {
         account,
         props.rootHostedZoneDNSName
       );
+      this.stagesHostedZones.push(stageSubZone);
```
Comment (jeromevdl) [inline source/aws-bootstrap-kit/lib/dns.ts:51]: only for tests?

File: source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts [hunk 3]
```diff
@@ -250,6 +253,20 @@ test("should not create any zone if no domain is provided", () => {
   expect(awsOrganizationsStack).toCountResources("AWS::Route53::HostedZone",0);
```
Comment (jeromevdl) [inline source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts:253]: not sure to understand why we have 0 in that case?

Comment (flochaz) [reply source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts:253]: if you don't give a domain we assume you don't want to get DNS management in Bootstrap Kit

File: source/aws-bootstrap-kit/lib/dns.ts (hunk 2 above)
Comment (flochaz) [reply source/aws-bootstrap-kit/lib/dns.ts:51]: What do you mean ? just populating the exposed array for potential reused by customer

File: source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts (hunk 1 above)
Comment (flochaz) [reply source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts:267]: sure will do

File: source/aws-bootstrap-kit/lib/dns.ts (hunk 2 above)
Comment (jeromevdl) [reply source/aws-bootstrap-kit/lib/dns.ts:51]: OK

File: source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts (hunk 3 above)
Comment (jeromevdl) [reply source/aws-bootstrap-kit/test/aws-organizations-stack.test.ts:253]: sorry, misread because of github collapses...

**Review by jeromevdl — APPROVED** [review]

**Pull request merged**